lib/state/state.js

```diff
@@ -439,75 +433,45 @@ State.prototype.emit = function (actorIds, path, data, language, isJson) {
 
 		sent[actorId] = true;
 
-		// language matches will be sorted out when we really emit them
-
-		var entry = { evt: evt, language: language };
-
-		if (!this.otherEvents) {
-			this.otherEvents = {};
-			this.otherEvents[actorId] = [entry];
-		} else {
-			if (!this.otherEvents[actorId]) {
-				this.otherEvents[actorId] = [entry];
-			} else {
-				this.otherEvents[actorId].push(entry);
-			}
+		if (!this.otherEvents[actorId]) {
+			this.otherEvents[actorId] = [];
 		}
+
+		this.otherEvents[actorId].push(entry);
 	}
 };
 
-// deprecated
-State.prototype.emitToActors = State.prototype.emit;
-
 
 State.prototype.error = function (code, logDetails, cb) {
 	exports.emit('stateError');
 
-	if (code === null || code === undefined) {
-		code = 'server';
-	}
-
-	if (code.code || code.message) {
-		// probably an error object, so turn it into a string
+	// Extract the error code or message
+	if (code instanceof Error) {
 		code = code.code || code.message;
 	}
 
-	if (typeof code === 'object' || Array.isArray(code)) {
+	// Invalid codes are logged, but a different value is set to be returned
+	if (!code || typeof code === 'object') {
 		logger.error('Invalid state error code:', code, '(falling back to "server")');
 		code = 'server';
 	}
 
+	// Code must be a string
 	code = '' + code;
 
-	var channel, args = [];
-
-	var prefix = [];
-
-	if (this.actorId) {
-		prefix.push('Actor ' + this.actorId);
-	}
-
-	if (this.description) {
-		prefix.push(this.description);
-	}
-
-	if (prefix.length > 0) {
-		args.push('(' + prefix.join(', ') + ')');
-	}
-
-	if (logDetails) {
-		channel = 'error';
-		args.push(logDetails);
-	} else {
-		channel = 'debug';
-		args.push('(error code: ' + JSON.stringify(code) + ')');
-	}
-
-	logger[channel].apply(logger, args);
-
+	// Return error code is set on the state
 	this.errorCode = JSON.stringify(code);
 
-	if (cb && typeof cb === 'function') {
+	// Log the result
+	logger.error
+		.details(logDetails)
+		.data({
+			actorId: this.actorId,
+			description: this.description
+		})
+		.log('state error');
```

Comment:
This is quite a breaking change, and not one I'm really happy with. Let's discuss offline.